filebeat/tests/system/test_registrar.py

```diff
@@ -161,6 +162,120 @@ def test_custom_registry_file_location(self):
+        with open(testfile_path, 'w') as testfile:
+            testfile.write("hello world\n")
+        filebeat = self.start_beat()
+        self.wait_until(
+            lambda: self.output_has(lines=1),
+            max_timeout=15)
+        # wait until the registry file exist. Needed to avoid a race between
+        # the logging and actual writing the file. Seems to happen on Windows.
+        self.wait_until(
+            lambda: os.path.isfile(os.path.join(self.working_dir,
+                                                "a/b/c/registry")),
+            max_timeout=1)
+        filebeat.check_kill_and_wait()
+
+        registry_file_perm_mask = oct(stat.S_IMODE(os.lstat(os.path.join(self.working_dir,
+                                                                         "a/b/c/registry")).st_mode))
+        self.assertEqual(registry_file_perm_mask, "0644")
+
+    def test_registry_file_update_permissions(self):
+        """
+        Test that filebeat registry permission is updated along with configuration
+        """
+
+        self.render_config_template(
+            path=os.path.abspath(self.working_dir) + "/log/*",
+            registryFile="a/b/c/registry_x",
+        )
+        os.mkdir(self.working_dir + "/log/")
+        testfile_path = self.working_dir + "/log/test.log"
+        with open(testfile_path, 'w') as testfile:
+            testfile.write("hello world\n")
+        filebeat = self.start_beat()
+        self.wait_until(
+            lambda: self.output_has(lines=1),
+            max_timeout=15)
+        # wait until the registry file exist. Needed to avoid a race between
+        # the logging and actual writing the file. Seems to happen on Windows.
+        self.wait_until(
+            lambda: os.path.isfile(os.path.join(self.working_dir,
+                                                "a/b/c/registry_x")),
+            max_timeout=1)
+        filebeat.check_kill_and_wait()
+
+        registry_file_perm_mask = oct(stat.S_IMODE(os.lstat(os.path.join(self.working_dir,
+                                                                         "a/b/c/registry_x")).st_mode))
+        self.assertEqual(registry_file_perm_mask, "0600")
+
+        self.render_config_template(
+            path=os.path.abspath(self.working_dir) + "/log/*",
+            registryFile="a/b/c/registry_x",
+            registryFilePermissions=0644
+        )
+
+        filebeat = self.start_beat()
+        self.wait_until(
+            lambda: self.output_has(lines=1),
+            max_timeout=15)
+        # wait until the registry file exist. Needed to avoid a race between
+        # the logging and actual writing the file. Seems to happen on Windows.
+        self.wait_until(
+            lambda: os.path.isfile(os.path.join(self.working_dir,
+                                                "a/b/c/registry_x")),
+            max_timeout=1)
+
+        # Wait a momemt to make sure registry is completely written
+        time.sleep(5)
```

Comment:
It would be nice if you could do a follow up PR that does not use `time.Sleep` but for example watches the registry file for the file permission changes. Otherwise this increases our CI time by at least 5 seconds. If we have this in many tests this is not nice.

Reply:
I should be able to work on a filebeat bug over the weekend. Will push it along with that.

Reply:
Great, but please make it 2 different PR's to sepearate concerns.